satellite/satellitedb/overlaycache.go

```diff
@@ -459,13 +519,36 @@ func (cache *overlaycache) UpdateStats(ctx context.Context, updateReq *overlay.U
 	return getNodeStats(dbNode), Error.Wrap(tx.Commit())
 }
 
-// UpdateOperator updates the email and wallet for a given node ID for satellite payments.
-func (cache *overlaycache) UpdateOperator(ctx context.Context, nodeID storj.NodeID, operator pb.NodeOperator) (stats *overlay.NodeDossier, err error) {
+// UpdateNodeInfo updates the email and wallet for a given node ID for satellite payments.
+func (cache *overlaycache) UpdateNodeInfo(ctx context.Context, nodeID storj.NodeID, nodeInfo *pb.InfoResponse) (stats *overlay.NodeDossier, err error) {
 	defer mon.Task()(&ctx)(&err)
 
-	updateFields := dbx.Node_Update_Fields{
-		Wallet: dbx.Node_Wallet(operator.GetWallet()),
-		Email:  dbx.Node_Email(operator.GetEmail()),
+	var updateFields dbx.Node_Update_Fields
+	if nodeInfo != nil {
+		if nodeInfo.GetOperator() != nil {
+			updateFields.Wallet = dbx.Node_Wallet(nodeInfo.GetOperator().GetWallet())
+			updateFields.Email = dbx.Node_Email(nodeInfo.GetOperator().GetEmail())
+		}
+		if nodeInfo.GetCapacity() != nil {
+			updateFields.FreeDisk = dbx.Node_FreeDisk(nodeInfo.GetCapacity().GetFreeDisk())
+			updateFields.FreeBandwidth = dbx.Node_FreeBandwidth(nodeInfo.GetCapacity().GetFreeBandwidth())
+		}
+		if nodeInfo.GetVersion() != nil {
+			semVer, err := version.NewSemVer(nodeInfo.GetVersion().GetVersion())
+			if err != nil {
+				semVer = &version.SemVer{}
```

Comment:
We should return the error here. Otherwise, it can result into hard to identify bugs.